clients/fides-js/src/lib/tcf.ts

```diff
@@ -192,30 +194,29 @@ export const initializeTcfCmpApi = () => {
     },
   });
 
-  // `null` value indicates that GDPR does not apply
   // Initialize api with TC str, we don't yet show UI, so we use false
   // see https://github.com/InteractiveAdvertisingBureau/iabtcf-es/tree/master/modules/cmpapi#dont-show-ui--tc-string-does-not-need-an-update
   window.addEventListener("FidesInitialized", (event) => {
     const tcString = fidesEventToTcString(event);
-    cmpApi.update(tcString ?? null, false);
```

Comment:
`fidesEventToTcString` now does the work of defaulting to a value. since if there's a tc string at all (including an empty string) the library sets `gdprApplies = true`. so `fidesEventToTcString` additionally checks if gdpr applies first. this might not actually be what we want though since it can effectively turn off setting a tc string at all. perhaps the override should just be for the initial default value? I did this at first, but I couldn't actually test it since the period where there's a default value until this line was too short for cypress to catch

Reply:
maybe only `FidesInitialized` should do the check for `fides_tcf_gdpr_applies` and the other events should stick to what they were?

Reply:
Hmmm... a philosophical question eh?

I would recommend we check for that option override in _one_ place, so I think it's OK to centralize into the helper function that extracts the string.

It's a bit weird that the `CmpApi` class sets GDPR applies based on a string though. Looking at their docs a bit to form a stronger POV

Reply:
OK, their docs are super explicit: https://github.com/InteractiveAdvertisingBureau/iabtcf-es/blob/master/modules/cmpapi/README.md#trigger-change-event
> update() may be called either an encoded [TC string](https://github.com/InteractiveAdvertisingBureau/GDPR-Transparency-and-Consent-Framework/blob/master/TCFv2/IAB%20Tech%20Lab%20-%20Consent%20string%20and%20vendor%20list%20formats%20v2.md#creating-a-tc-string) an empty string ('') or null.
> - Encoded TC string, CmpApi will decode the string and respond to [TCData](https://github.com/InteractiveAdvertisingBureau/GDPR-Transparency-and-Consent-Framework/blob/master/TCFv2/IAB%20Tech%20Lab%20-%20CMP%20API%20v2.md#tcdata) with the decoded values.
>    - gdprApplies will be set to true
> - Empty string (''), CmpApi will respond to [TCData](https://github.com/InteractiveAdvertisingBureau/GDPR-Transparency-and-Consent-Framework/blob/master/TCFv2/IAB%20Tech%20Lab%20-%20CMP%20API%20v2.md#tcdata) with the correct structure but all primitive values will be empty.
>    - gdprApplies will be set to true
> - null, CmpApi will respond to [TCData](https://github.com/InteractiveAdvertisingBureau/GDPR-Transparency-and-Consent-Framework/blob/master/TCFv2/IAB%20Tech%20Lab%20-%20CMP%20API%20v2.md#tcdata) with the correct structure but all primitive values will be empty.
>    - gdprApplies will be set to false

In that case I think we can document this more clearly but I agree with your implementation 👍